src/main/java/bio/terra/service/dataset/exception/FatalQueryException.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.service.dataset.exception;
```

Comment:
This seems like it is not dataset-specific. Could you move it in to the common package instead?